Library/Homebrew/extend/ENV/std.rb

```diff
@@ -28,7 +28,8 @@ def setup_build_environment(formula: nil, cc: nil, build_bottle: false, bottle_a
 
     self["HOMEBREW_ENV"] = "std"
 
-    PATH.new(ENV["HOMEBREW_PATH"]).each { |p| prepend_path "PATH", p }
+    PATH.new(ENV["HOMEBREW_PATH"]).reverse_each { |p| prepend_path "PATH", p }
```

Comment:
Is there a reason this is reversed now or just for consistency between `HOMEBREW_PATH` and `PATH`?

Maybe I'm misunderstanding but this feels like the kind of thing that could have an unexpectedly huge impact (e.g. anytime two executables with the same name are present).

Reply:
What this was doing before was going reading HOMEBREW_PATH from left to right, but adding them to PATH from right to left.

The change means that the order of the PATH now matches exactly as it was in HOMEBREW_PATH - which I see as more expected behaviour.

Reply:
I agree that this makes more sense, but I'm just concerned that making the change will have unexpected consequences. If you don't think it will cause problems, 👍 from me.

Reply:
I think it's one of those things where if they are relying on this, they are relying on a bug and it wouldn't work outside of the Homebrew environment since the PATH outside would be the reverse.

Reply:
Fair. We can always switch back if there are problems